Separate documentation for WebGL 1 and 2 methods/properties (#9699)
mdn/content: files/en-us/web/api/webgl2renderingcontext/bufferdata/index.md

---
title: WebGL2RenderingContext.bufferData()
slug: Web/API/WebGL2RenderingContext/bufferData
tags:
  - API
  - Buffer
  - Graphics
  - Method
  - Reference
  - WebGL
  - WebGL2RenderingContext
browser-compat: api.WebGL2RenderingContext.bufferData
---
{{APIRef("WebGL2")}}

> **Note:** This page describes the WebGL 2 version of this method.  Click [here](/en-US/docs/Web/API/WebGLRenderingContext/bufferData) to read about the WebGL 1 version.

The **`WebGL2RenderingContext.bufferData()`** method of the [WebGL API](/en-US/docs/Web/API/WebGL_API) initializes and creates the
buffer object's data store.

## Syntax

```js

> void gl.bufferData(target, ArrayBufferView srcData, usage, srcOffset, length);

Reviewer: Should this also include all of the signatures that work in WebGL 1, and all the variants you get due to some parameters being optional?

Also, should the type `ArrayBufferView` be in syntax blocks?

Reviewer 2: > Should this also include all of the signatures that work in WebGL 1, and all the variants you get due to some parameters being optional?

Yes, because they all work in the WebGL 2 version of this method.

> Also, should the type ArrayBufferView be in syntax blocks?

No, but this seems to be copied from the original page where it is wrong already.

---
title: WebGL2RenderingContext.bufferData()
slug: Web/API/WebGL2RenderingContext/bufferData
tags:
  - API
  - Buffer
  - Graphics
  - Method
  - Reference
  - WebGL
  - WebGL2RenderingContext
browser-compat: api.WebGL2RenderingContext.bufferData
---
{{APIRef("WebGL2")}}

> > **Note:** This page describes the WebGL 2 version of this method.  Click [here](/en-US/docs/Web/API/WebGLRenderingContext/bufferData) to read about the WebGL 1 version.

Reviewer 2: "Click here" is something we try to avoid.

```suggestion
> **Note:** This page describes the WebGL 2 version of this method. [`WebGLRenderingContext.bufferData()`](/en-US/docs/Web/API/WebGLRenderingContext/bufferData) is the WebGL 1 version of this method.
```
